File: Algorithms_Java/binarySearch.java
package Algorithms_Java;
public class binarySearch {

    public static int bS(int[] arr, int find){
        int left=0;
        int right = arr.length-1;

        while(left <= right){
            int mid = left + (right - left) / 2;

            if (arr[mid] == find){
                return mid;
            } else if (arr[mid] < find){
                left = mid + 1;
            } else{
                right = mid - 1;
            }
        }
        return -1;
    }

    public static int bsc(int[] arr, int left, int right, int find){
        if (left > right){
            return -1;
        }
        
        int mid = left + (right - left) / 2;

        if (arr[mid] == find){
            return mid;
        }else if (arr[mid] < find){
            return bsc(arr, mid + 1, right, find);
        }else{
            return bsc(arr, left, mid -1, find);
        }
    }

    public static void main(String[] args){
        int[] arr = {1, 3, 7, 11, 17, 23, 29, 40, 45};
        int index = bsc(arr, 0, arr.length-1, 239489);
        System.out.print(index);
    }
}

File: Algorithms_Cpp/binarySearch.cpp
#include <iostream>
#include <queue>
#include <vector>
#include <climits>
using namespace std;

int binarySearch(vector<int>& arr, int l, int r, int find){
    if(l<=r){
        int m = l + (r-l)/2;
        if(arr[m] == find){
            return m;
        }else if(arr[m] < find){
            return binarySearch(arr, m+1, r, find);
        }else{
            return binarySearch(arr, l, m-1, find);
        }
    }    
    return -1;   
} 

int main(){
    vector<int> arr = {1, 2, 3, 4, 5, 6, 7, 8, 9};
    cout << binarySearch(arr, 0, arr.size(), 4)+1;
    return 0;
}